src/spatialdata/_core/query/spatial_query.py

```diff
@@ -74,38 +75,65 @@ def _get_bounding_box_corners_in_intrinsic_coordinates(
 
     The axes of the intrinsic coordinate system.
     """
-    from spatialdata.transformations import get_transformation
-
     min_coordinate = _parse_list_into_array(min_coordinate)
     max_coordinate = _parse_list_into_array(max_coordinate)
-    # get the transformation from the element's intrinsic coordinate system
-    # to the query coordinate space
-    transform_to_query_space = get_transformation(element, to_coordinate_system=target_coordinate_system)
+
+    # compute the output axes of the transformation, remove c from input and output axes, return the matrix without c
+    # and then build an affine transformation from that
     m_without_c, input_axes_without_c, output_axes_without_c = _get_axes_of_tranformation(
         element, target_coordinate_system
     )
+    spatial_transform = Affine(m_without_c, input_axes=input_axes_without_c, output_axes=output_axes_without_c)
+
+    # we identified 5 cases (see the responsible function for details), cases 1 and 5 correspond to invertible
+    # transformations; we focus on them
+    if isinstance(element, (DataArray, DataTree)):
+        m_without_c_linear = m_without_c[:-1, :-1]
+        case = _get_case_of_bounding_box_query(m_without_c_linear, input_axes_without_c, output_axes_without_c)
+        assert case in [1, 5]
```

Comment:
we should replace this `assert` with `if ... raise`, so that this is triggered even when running `python` with `-O`

Reply:
(I am the one to (git) blame for having used the asserts in the first place 😅)

Reply:
ok, I actually noticed that it is raised already as an error here so I would remove this check entirely https://github.com/scverse/spatialdata/blob/cc85a7fca124bfb205021cee33b275419fc0fae0/src/spatialdata/_core/query/spatial_query.py#L308-L316